modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-container-test/src/testIntegration/java/com/liferay/portal/osgi/web/portlet/container/test/RenderRequestPortletContainerTest.java

```diff
@@ -91,7 +91,7 @@ public void testInvalidPortletId() throws Exception {
 				captureAppender.getLoggingEvents();
 
 			Assert.assertEquals(
-				loggingEvents.toString(), 1, loggingEvents.size());
+				loggingEvents.toString(), 2, loggingEvents.size());
```

Comment:
@ngriffin7a don't just bump the expected log size, we should assert what it in the new log too.

Reply:
@shuyangzhou I changed `1` to `2` on this line in RenderRequestPortletContainerTest.java:
```
Assert.assertEquals(
	loggingEvents.toString(), 2, loggingEvents.size());
```
The test tries to hit this URL:
```
http://localhost:8080/web/guest/home?p_p_id=%27%22%3E%3Cscript%3Ealert%281%29%3C%2Fscript%3E&p_p_lifecycle=0&p_p_state=exclusive
```

Which of course has an invalid `p_p_id` value. This causes the following output to appear in the console log:

```
2018-05-25 18:53:55.161 WARN  [http-nio-8080-exec-2][SecurityPortletContainerWrapper:227] Invalid portlet ID '"><script>alert(1)</script>
2018-05-25 18:53:55.964 WARN  [http-nio-8080-exec-2][SecurityPortletContainerWrapper:227] Invalid portlet ID '"><script>alert(1)</script>
```

The reason why the WARN used to appear once in the log, and now it appears twice, is because `checkRender(request, portlet)` is called by the new `SecurityPortletContainerWrapper.renderHeaders(...)` method as well as the existing `SecurityPortletContainerWrapper.render(...)` method.

So I felt that the test should be corrected according to the new behavior.

Reply:
@ngriffin7a I don't mind how many logs has been generated, but we need to assert what we are capturing. Otherwise we are losing the point for capturing the logs.